Library/Homebrew/keg.rb

```diff
@@ -101,6 +102,19 @@ def initialize(path)
     @name = path.parent.basename.to_s
     @linked_keg_record = HOMEBREW_LIBRARY.join("LinkedKegs", name)
     @opt_record = HOMEBREW_PREFIX.join("opt", name)
+    begin
+      tab = Tab.from_file(path/"INSTALL_RECEIPT.json")
+      tap = tab.tap
+
+      oldname = if tap.nil? || tap == "Homebrew/homebrew"
+        Formulary.factory(name)
+      else
+        Formulary.factory("#{tap.sub("homebrew-", "")}/#{name}")
+      end.oldname
+
+      @oldname_opt_record = HOMEBREW_PREFIX.join("opt", oldname) if oldname
+    rescue Exception
+    end
```

Comment:
I'd like to avoid loading formula here. Instead we can just get it by reading the filesystem. Beside this, in order to improve the performance, let's make it a separate method instead of computing it in `initialize`. So in all, here is my proposal.

```ruby
def oldname_opt_record
  @oldname_opt_record ||= if (opt_dir = HOMEBREW_PREFIX/"opt").directory?
    opt_dir.subdirs.detect { |dir| dir.symlink? && path == dir.resolved_path }
  end
end

def remove_oldname_opt_record
  oldname_opt_record.unlink
  oldname_opt_record.parent.rmdir_if_possible
  @oldname_opt_record = nil
end
```